engine/apps/mobile_app/fcm_relay.py

```diff
@@ -42,18 +42,19 @@ def post(self, request):
             token = request.data["token"]
             data = request.data["data"]
             apns = request.data["apns"]
+            android = request.data.get("android")  # optional
         except KeyError:
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
-        fcm_relay_async.delay(token=token, data=data, apns=apns)
+        fcm_relay_async.delay(token=token, data=data, apns=apns, android=android)
         return Response(status=status.HTTP_200_OK)
 
 
 @shared_dedicated_queue_retry_task(
     autoretry_for=(Exception,), retry_backoff=True, max_retries=1 if settings.DEBUG else 5
 )
-def fcm_relay_async(token, data, apns):
-    message = Message(token=token, data=data, apns=deserialize_apns(apns))
+def fcm_relay_async(token, data, apns, android=None):
+    message = _get_message_from_request_data(token, data, apns, android)
```

Comment:
do we need the `android` argument here? this field should be available on the `FCMDevice` object fetched just a few lines below. See the `type` column in the `fcm_django_fcmdevice` table.